tests/php/Publisher/FilesystemPublisherTest.php

```diff
@@ -193,70 +195,80 @@ public function testOnlyHTML()
         $level1->write();
         $level1->publishRecursive();
 
+        $this->logOut();
+
         $this->fsp->publishURL($level1->Link(), true);
         $staticFilePath = $this->fsp->getDestPath() . 'mimetype';
 
         $this->assertFileExists($staticFilePath . '.html');
-        $this->assertFileNotExists($staticFilePath . '.php');
-        $this->assertSame(
+        $this->assertFileDoesNotExist($staticFilePath . '.php');
+        $this->assertEquals(
             '<div class="statically-published" style="display: none"></div>',
             trim(file_get_contents($staticFilePath . '.html'))
         );
     }
 
-    public function testPurgeURL()
+    public function testPurgeURL(): void
     {
         $level1 = new StaticPublisherTestPage();
         $level1->URLSegment = 'to-be-purged';
         $level1->write();
         $level1->publishRecursive();
 
+        $this->logOut();
+
         $this->fsp->publishURL('to-be-purged', true);
         $this->assertFileExists($this->fsp->getDestPath() . 'to-be-purged.html');
         $this->assertFileExists($this->fsp->getDestPath() . 'to-be-purged.php');
 
         $this->fsp->purgeURL('to-be-purged');
-        $this->assertFileNotExists($this->fsp->getDestPath() . 'to-be-purged.html');
-        $this->assertFileNotExists($this->fsp->getDestPath() . 'to-be-purged.php');
+        $this->assertFileDoesNotExist($this->fsp->getDestPath() . 'to-be-purged.html');
+        $this->assertFileDoesNotExist($this->fsp->getDestPath() . 'to-be-purged.php');
     }
 
-    public function testPurgeURLAfterSwitchingExtensions()
+    public function testPurgeURLAfterSwitchingExtensions(): void
     {
         $level1 = new StaticPublisherTestPage();
         $level1->URLSegment = 'purge-me';
         $level1->write();
         $level1->publishRecursive();
 
+        $this->logOut();
```

Comment:
Here is another solution for the case if you specifically want to test the case when a user is logged in.

```php
// Make sure that we retain Cookie_Backend service as @see Publisher::getHTTPApplication()
// creates a fresh Injector instance throwing away any Injector level configuration in the process
Injector::inst()->registerService(new CookieJarMock(), Cookie_Backend::class);
```

```php
<?php

namespace App\Tests\StaticPublishing;

use SilverStripe\Control\CookieJar;
use SilverStripe\Dev\TestOnly;

/**
 * Test only version of @see CookieJar to provide ability to skip cookies output as it may cause unit test failures
 * for tests which execute output rendering such as static cache jobs
 */
class CookieJarMock extends CookieJar implements TestOnly
{
    public function set($name, $value, $expiry = 90, $path = null, $domain = null, $secure = false, $httpOnly = true)
    {
        // No op
    }

    public function forceExpiry($name, $path = null, $domain = null, $secure = false, $httpOnly = true)
    {
        // No op
    }
}
```

Reply:
Thanks, @mfendeksilverstripe!

Would I add that `Injector` portion into our existing `StaticPublisherState` (`TestState`)?

Reply:
Hmmmm, that didn't seem to work. I think I'll just stick with the `logOut()` for now. Adding the Mock can be an improvement if it's needed down the line.